echoserver/EchoServer.java

```diff
@@ -0,0 +1,136 @@
+package com.gymfox.echoserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class EchoServer {
+    private final int port;
+    private volatile boolean isRunning;
+    private BufferedReader inputMessage;
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private final static int DEFAULT_PORT = 8080;
+    private final static int MIN_PERMISSION_PORT = 1024;
+    private final static int MAX_PERMISSION_PORT = 65536;
+    private ExecutorService pool = Executors.newFixedThreadPool(2);
+
+    public static class InvalidPortException extends Exception{}
+
+    public EchoServer() throws InvalidPortException {
+        this(DEFAULT_PORT);
+    }
+
+    public EchoServer(int port) throws InvalidPortException {
+        validate(port);
+        this.port = port;
+        this.isRunning = false;
+    }
+
+    private void validate(int port) throws InvalidPortException {
+        if ( port < MIN_PERMISSION_PORT || port > MAX_PERMISSION_PORT ) {
+            throw new InvalidPortException();
+        }
+    }
+
+    public boolean isRunning() {
+        return isRunning;
+    }
+
+    public int getPort() {
+        return port;
+    }
+
+    public void start() throws IOException {
+        run();
+
+        while (isRunning()) {
+            clientSocket = serverSocket.accept();
+            System.out.println("Client has been connected\n");
+
+            pool.execute(() -> {
+                try {
+                    while ( clientIsNotClose() ) {
+                        inputMessage = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
+                        String line = inputMessage.readLine();
```

Comment:
It would be good to have a timeout for the socket, regarding that `readLine()` is blocking and so a client can block execution thread forever if won't print anything. Consider `Socket::setToTimeout` method`